mapstruct-on-gradle-testcp-with-lombok/build.gradle

```diff
@@ -0,0 +1,25 @@
+plugins {
+    id 'java'
+    id "io.freefair.lombok" version "6.0.0-m2"
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.7.0'
+    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.7.0'
+
+    annotationProcessor 'org.mapstruct:mapstruct-processor:1.4.2.Final'
+    compileOnly 'org.mapstruct:mapstruct-processor:1.4.2.Final'
```

Comment:
Why do we need mapStruct on the `compile` class path?

Reply:
This is adressing an issue/bug in intellij, for gradle, the annotationProcessor is fine. Gradle will not add transitive dependencies on compileOnly, so it’s not a big issue. But I‘m fine with removing these compileOnly.

Reply:
It is not for the transitive dependencies, but more that users might accidentally use classes from the MapStruct processor. If it works without it, let's remove it